From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Thu, 22 Dec 2016 22:04:42 +0100
Subject: [PATCH 12/19] init.rc: don't start console unless asked to

This can have a major impact on performance.

We don't want this, even on userdebug/eng builds.

Use the new property "ro.console.enable" to
enable the console service explicitly.

Change-Id: I93e7c65e92261443d1c9c70cfef9aa2ed5ff328a
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
Signed-off-by: Jason Edson <jaysonedson@gmail.com>
Signed-off-by: Dmitrii <bankersenator@gmail.com>
Signed-off-by: Joey Huab <joey@evolution-x.org>

Former-commit-id: 4d36ac6b9fe0a26dcfa4fc32de377c502f49eb3e
Former-commit-id: 1dc24bca6c98f3a9a23ce5ec3bf76c2019973694
Former-commit-id: c31e332873ff454d4203da5c67188a79260d2c1c
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 92dcdd0..c6a517f 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -1364,7 +1364,7 @@ on property:ro.debuggable=1
     # Give reads to anyone for the accessibility trace folder on debug builds.
     chmod 0775 /data/misc/a11ytrace
 
-on init && property:ro.debuggable=1
+on init && property:ro.debuggable=1 && property:ro.console.enable=1
     start console
 
 # Multi-Gen LRU Experiment
